app/views/header.js

```diff
@@ -0,0 +1,56 @@
+import Mn from 'backbone.marionette';
+import application from '../application';
+import Tracks from '../collections/tracks';
+
+const Header = Mn.ItemView.extend({
+
+    template: require('./templates/header'),
+
+    events: {
+        'click #reset-upnext': 'resetUpNext',
+        'click #delete-playlist': 'deletePlaylist'
+    },
+
+    modelEvents: { "change:currentPlaylist": 'changedPlaylist' },
+
+    initialize() {
+       this.listenTo(
+            this.model.get('currentPlaylist').get('tracks'),
+            'update reset',
+            this.render
+        );
+   },
+
+    serializeData() {
+        let currentPlaylist = this.model.get('currentPlaylist');
+        return {
+            title: currentPlaylist.get('title'),
+            count: currentPlaylist.get('tracks').length
+        }
+    },
+
+    changedPlaylist(model, newValue) {
+        this.stopListening(model.changed.currentPlaylist);
+        this.listenTo(
+            this.model.get('currentPlaylist').get('tracks'),
+            'update reset',
+            this.render
+        );
+        this.render();
+    },
+
+    deletePlaylist() {
```

Comment:
Ok, that's what I understood. I think it's not the reponsability of the `header` view to do it. I suggest it trigger a `request` in the app.channel, and the `allPlaylists` collection listen to this event to delete itself the element. I loose the coupling and let us add other methods to do the same thing later (e.g. a trash button that appear when hovering the playlist name in the drawer).